src/app/shared/services/version.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { HttpService } from './http.service';
+import { firstValueFrom } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class VersionService {
+  packageJsonPath = 'assets/package.json';
+  frontendVersion?: string;
+  backendVersion?: string;
+
+  constructor(
+    private httpService: HttpService,
+    private httpClient: HttpClient,
+  ) {}
+
+  async getFrontendVersion() {
+    if (!this.frontendVersion) {
+      try {
+        const packageJson = await firstValueFrom(this.httpClient.get<{ version: string }>(this.packageJsonPath));
+        if (packageJson) {
+          this.frontendVersion = packageJson.version;
+        }
+      } catch (error) {
+        console.error('package.json could not be found in assets', error);
```

Comment:
If `this.frontendVersion` is undefined it will never in this catch, shouldn't you set it to a default value?
This will break other components that also have set the frontendVersion variable to be able to be undefined but is used within a template